src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -554,3 +558,59 @@ export const updateFormSettings = (
     return okAsync(updatedForm.getSettings())
   })
 }
+
+/**
+ * Deletes form logic.
+ * @param form The original form to delete logic in
+ * @param logicId the logicId to delete
+ * @returns ok(true) on success
+ * @returns err(database errors) if db error is thrown during logic delete
+ * @returns err(LogicNotFoundError) if logicId does not exist on form
+ */
+export const deleteFormLogic = (
+  form: IPopulatedForm,
+  logicId: string,
+): Result<never, LogicNotFoundError> | ResultAsync<true, DatabaseError> => {
+  // First check if specified logic exists
+  if (!form.form_logics.some((logic) => logic.id === logicId)) {
+    logger.error({
+      message: 'Error occurred when deleting form logic',
+      meta: {
+        action: 'deleteFormLogic',
+        formId: form._id,
+        logicId,
+      },
+    })
+    return err(new LogicNotFoundError('logicId does not exist on form'))
+  }
+
+  // Remove specified logic and then update form logic
+  const updated_form_logic = form.form_logics.filter(
+    (logic) => logic.id !== logicId,
+  )
+  const ModelToUse = getFormModelByResponseMode(form.responseMode)
```

Comment:
I think general form model would work, don't need to retrieve form model by response mode right?

Reply:
edited